gradle-module-plugin/src/main/kotlin/io/ia/sdk/gradle/modl/task/SignModule.kt

```diff
@@ -208,10 +284,44 @@ open class SignModule @Inject constructor(_providers: ProviderFactory, _objects:
             logger.debug("Found unsigned module at ${unsignedModule.absolutePath}...")
         }
 
+        // If both file- and PKCS#11-based keystores are specified, fail.
+        // Mechanically that means these properties are both 1) set, 2) set
+        // to something other than the SKIP value, and 3) not explicitly set
+        // to null via property convention backstop logic.
+        if (
+            listOf(keystorePath, pkcs11CfgPath).all { p ->
+                listOf(SKIP, null).none { v -> p.getOrNull() == v }
+            }
+        ) {
+            throw InvalidUserDataException(
```

Comment:
If you squint at the Gradle API doc, this is the closest of their exception types that matches this situation.